cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -436,8 +459,11 @@ func (g *gitHubRepository) httpGetFilesFromRelease(ctx context.Context, version,
 
 // downloadFilesFromRelease download a file from release.
 func (g *gitHubRepository) downloadFilesFromRelease(ctx context.Context, release *github.RepositoryRelease, fileName string) ([]byte, error) {
-	client := g.getClient()
 	absoluteFileName := filepath.Join(g.rootPath, fileName)
+	client, err := g.getClient()
+	if err != nil {
```

Comment:
Make sense to check for an error here, and it's a bit odd that we weren't returning before. Maybe someone else can chime in on if there's a reason we would proceed with an error.